src/shared_modules/indexer_connector/tests/unit/monitoring_test.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Wazuh Indexer Connector - Monitoring tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 08, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _MONITORING_TEST_HPP
+#define _MONITORING_TEST_HPP
+
+#include "fakeOpenSearchServer.hpp"
+#include "monitoring.hpp"
+#include <gtest/gtest.h>
+#include <memory>
+#include <string>
+#include <vector>
+
+/**
+ * @brief Runs unit tests for Monitoring class
+ */
+class MonitoringTest : public ::testing::Test
+{
+protected:
+    inline static std::unique_ptr<FakeOpenSearchServer>
+        m_fakeOpenSearchGreenServer; ///< pointer to FakeOpenSearchServer class
+
+    inline static std::unique_ptr<FakeOpenSearchServer>
+        m_fakeOpenSearchRedServer; ///< pointer to FakeOpenSearchServer class
+
+    std::shared_ptr<Monitoring> m_monitoring; ///< pointer to Monitoring class
+
+    std::vector<std::string> m_servers; ///< Servers
+
+    /**
+     * @brief Sets initial conditions for each test case.
+     */
+    // cppcheck-suppress unusedFunction
+    void SetUp() override
+    {
+        // Register the host and port of the green server
+        m_servers.emplace_back("http://localhost:9201");
+        // Register the host and port of the red server
+        m_servers.emplace_back("http://localhost:9202");
```

Comment:
Nitpick: In alignment with the other proposal. This should be a fixed size array of std::string_view

Reply:
It cannot be an array because classes `Monitoring` and `ServerSelector` have a vector as a parameter.

Reply:
You are right